source/mesh_deformation/fastscape.cc

```diff
@@ -0,0 +1,1739 @@
+                                          &gsed,
+                                          &p);
+
+      if (use_marine_component)
+        fastscape_set_marine_parameters_(&sea_level,
+                                         &p1,
+                                         &p2,
+                                         &z1,
+                                         &z2,
+                                         &r,
+                                         &l,
+                                         &kds1,
+                                         &kds2);
+
+      // Only set the basement and silt_fraction if it's a restart
+      if (current_timestep != 1)
+        {
+          fastscape_set_basement_(b.data());
+          if (use_marine_component)
+            fastscape_init_f_(sf.data());
+        }
+
+    }
+
+
+    template <int dim>
+    void FastScape<dim>::execute_fastscape(std::vector<double> &h,
+                                           std::vector<double> &kd) const
+    {
+      TimerOutput::Scope timer_section(this->get_computing_timer(), "Execute FastScape");
+      if (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) != 0)
+        return;
+
+      const double aspect_timestep_in_years = this->get_timestep() / year_in_seconds;
+
+      // Because on the first timestep we will create an initial VTK file before running FastScape
+      // and a second after, we first set the visualization step to zero.
+      int visualization_step = 0;
+      const unsigned int current_timestep = this->get_timestep_number ();
+      std::string dirname = (this->get_output_directory() + "fastscape/");
+      const char *c=dirname.c_str();
+      int length = dirname.length();
+
+      // Find a FastScape timestep that is below our maximum timestep.
+      unsigned int fastscape_iterations = fastscape_steps_per_aspect_step;
+      double fastscape_timestep_in_years = aspect_timestep_in_years/fastscape_iterations;
+      while (fastscape_timestep_in_years>maximum_fastscape_timestep)
+        {
+          fastscape_iterations *= 2;
+          fastscape_timestep_in_years *= 0.5;
+        }
+
+      // Set time step
+      fastscape_set_dt_(&fastscape_timestep_in_years);
+      this->get_pcout() << "   Executing FastScape... " << (fastscape_iterations) << " timesteps of " << fastscape_timestep_in_years << " years." << std::endl;
+      {
+        // If it is the first timestep, write an initial VTK file.
+        if (current_timestep == 1)
+          {
+            this->get_pcout() << "      Writing initial VTK..." << std::endl;
+            // Note: Here, the HHHHH field that is set to show the diffusivity (kd).
+            // However, you can change this so that any parameter is visualized.
+            fastscape_named_vtk_(kd.data(),
+                                 &vexp,
+                                 &visualization_step,
+                                 c,
+                                 &length);
```

Comment:
I've updated the comment to be more descriptive. The original fastscape_vtk function would output only one field called HHHHH, which would show whichever data you put into it. We updated the function to by default output quite a few different fields, but not all of them, so I had left the HHHHH field as is in case someone might want to visualize something else, but by default it is sent the bedrock diffusivity data.